AdobeStockClient/Test/Unit/Model/Client/Files/CurlResultSuccessFilesExistsTest.php

```diff
@@ -0,0 +1,142 @@
+use Magento\Framework\HTTP\Client\Curl;
+use Magento\Framework\HTTP\Client\CurlFactory;
+use Magento\Framework\Locale\Resolver;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test case when the curl files request is successful and files present in request
+ */
+class CurlResultSuccessFilesExistsTest extends TestCase
+{
+    /**
+     * @var ImsConfig|MockObject
+     */
+    private $imsConfigMock;
+
+    /**
+     * @var ClientConfig|MockObject
+     */
+    private $clientConfigMock;
+
+    /**
+     * @var Resolver|MockObject
+     */
+    private $localeResolverMock;
+
+    /**
+     * @var GetAccessTokenInterface|MockObject
+     */
+    private $getAccessTokenMock;
+
+    /**
+     * @var Files|MockObject
+     */
+    private $files;
+
+    /**
+     * @var Curl|MockObject
+     */
+    private $curlAdapterMock;
+
+    /**
+     * @var Json|MockObject
+     */
+    private $jsonMock;
+
+    /**
+     * Prepare test objects.
+     */
+    protected function setUp(): void
+    {
+        $this->imsConfigMock = $this->createMock(ImsConfig::class);
+        $this->clientConfigMock = $this->createMock(ClientConfig::class);
+        $this->localeResolverMock = $this->createMock(Resolver::class);
+        $this->getAccessTokenMock = $this->createMock(GetAccessTokenInterface::class);
+        $this->jsonMock = $this->createMock(Json::class);
+        $curlFactoryMock = $this->createMock(CurlFactory::class);
+
+        $this->files = (new ObjectManager($this))->getObject(
+            Files::class,
+            [
+                'imsConfig' => $this->imsConfigMock,
+                'clientConfig' => $this->clientConfigMock,
+                'localeResolver' => $this->localeResolverMock,
+                'getAccessToken' => $this->getAccessTokenMock,
+                'curlFactory' => $curlFactoryMock,
+                'json' => $this->jsonMock,
+            ]
+        );
+
+        $this->curlAdapterMock = $this->createMock(Curl::class);
+        $curlFactoryMock->expects($this->once())
+            ->method('create')
+            ->willReturn($this->curlAdapterMock);
+    }
+
+    /**
+     * Search test
+     */
+    public function testCurlResultSuccessFilesExist(): void
+    {
+        $ids = [1];
+        $columns = [];
+        $locale = 'en';
+        $files = ['files'];
+
+        $this->clientConfigMock->expects($this->once())
+            ->method('getProductName')
+            ->willReturn('string');
+
+        $this->imsConfigMock->expects($this->once())
+            ->method('getApiKey')
+            ->willReturn('string');
+
+        $this->getAccessTokenMock->expects($this->once())
+            ->method('execute')
+            ->willReturn('array');
+
+        $this->curlAdapterMock->expects($this->once())
+            ->method('setHeaders')
```

Comment:
I believe `->with(...)` on `setHeaders` method is one of the most important assertions in the test